### ISSUE TITLE
`Disk named `public` cannot be accessed`

### ISSUE THREAD

**Issue by Kokil** [spatie/laravel-medialibrary#3267]
I'm encountering an issue while using laravel-medialibrary version ^10.9 . 
When i run php artisan media-library:regenerate it give 
`Media id 9: `Disk named `public` cannot be accessed``

I have verified my filesystem configuration in config/filesystems.php and it appears to be set correctly:

```
'public' => [
    'driver' => 'local',
    'root' => storage_path('app/public'),
    'url' => env('APP_URL').'/storage',
    'visibility' => 'public',
    'throw' => false,
],
```
Could you please guide me in troubleshooting this issue? Is there anything specific I should be aware of when using laravel-medialibrary with Laravel 10.x?

**Comment by parallels999** [comment]
Folder permissions

**Comment by Kokil** [comment]
> Folder permissions

I don't think it's due to folder permissions. 
after your comment, I tried giving even 777 to the storage folder.

If I upload from the backend it's working.

**Comment by francoism90** [comment]
@Kokil What is your environment? Is this Laravel Sail or production?

**Comment by francoism90** [comment]
@Kokil I also think this is a permission issue.

**Comment by Kokil** [comment]
![image](https://github.com/spatie/laravel-medialibrary/assets/481786/b7ad8b67-b2ce-4974-83a1-19aaed72e867)

**Comment by francoism90** [comment]
I really don't know, if your public correctly defined in `config/filesystems.php`? Tried clearing your cache? Also does the symlink correctly exists?

Anyway, I don't think this is a package issue, sorry.

**Comment by Kokil** [comment]
After your comment, I take the following step :
1. Get the fresh filesystems.php.
2. I run artisan migrate:fresh --seed
3. cache: clear, remove old symlink, regenerate symlink
4. i upload the media from backend upload success.
5. after that i run : php artisan media-library:regenerate
again same
![image](https://github.com/spatie/laravel-medialibrary/assets/481786/f7e9b61c-d614-4304-93a9-412418719560)

PHP 8.1.7 & Laravel Framework 10.11.0
Note :
I migrate this application laravel 8 to laravel 10.
at laravel 8 i was working fine.

**Comment by bhojkamal** [comment]
I also got same error on laravel 10. but it was working in same way in laravel 9. The error is 
```
"message": "Disk named `public` cannot be accessed",
    "exception": "Spatie\\MediaLibrary\\MediaCollections\\Exceptions\\DiskCannotBeAccessed",
```

**Comment by Kokil** [comment]
Yes i also upgrade from laravel 8 in laravel 8 it was working fine.

**Comment by Kokil** [comment]
any update on this ?
